### PR TITLE
Rework all routing to use the new artificial key and partition number (IOW-562)

### PR DESCRIPTION
**Pull request by kkehl-usgs**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

Rework all routing and proprocessing takes to use the new artificial key and partition number
-----------
Add uuid column to json_data table.  Add index as well.

https://internal.cida.usgs.gov/jira/browse/IOW-562

Description
-----------

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

File: liquibase/changeLogs/capture/indexes/jsonData.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+databaseChangeLog:
+  - preConditions:
+    - dbms:
+        type: postgresql
+    - runningAs:
+        username: ${AQTS_SCHEMA_OWNER_USERNAME}
+
+  - changeSet:
+      author: ssoper
```
Comment (ssoper-usgs) [inline liquibase/changeLogs/capture/indexes/jsonData.yml]: Feel free to pop your credentials in here @kkehl-usgs !

File: liquibase/changeLogs/capture/tables/jsonData/changeLog.yml [hunk 2]
```diff
@@ -105,3 +105,18 @@ databaseChangeLog:
...
+      preConditions:
+        - onFail: MARK_RAN
+        - onError: HALT
+        - not:
+            - columnExists:
+                schemaName: ${AQTS_SCHEMA_NAME}
+                tableName: json_data
+                columnName: uuid
+      changes:
+        - sql: alter table ${AQTS_SCHEMA_NAME}.json_data add column if not exists uuid character(36);
```
Comment (ssoper-usgs) [inline liquibase/changeLogs/capture/tables/jsonData/changeLog.yml:121]: Are these always 36 characters long? Just curious.

**Review by ssoper-usgs — APPROVED** [review]

File: liquibase/changeLogs/capture/tables/jsonData/changeLog.yml (hunk 2 above)
Comment (kkehl-usgs) [reply liquibase/changeLogs/capture/tables/jsonData/changeLog.yml:121]: Think so.  The UUID4 format looks like:  16fd2706-8baf-433b-82eb-8c7fada847da.

Also, the example I found in S3 looks like:

body_getTSData_10000_mod_    +   0074973a-c377-472f-9f3c-f093fdc18836  + .json

**Review by ayan-usgs — APPROVED** [review]

File: liquibase/changeLogs/capture/indexes/jsonData.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        type: postgresql
+    - runningAs:
+        username: ${AQTS_SCHEMA_OWNER_USERNAME}
+
+  - changeSet:
+      author: kkehl
+      id: "create.index.${AQTS_SCHEMA_NAME}.json_data_uuid"
+      changes:
+        - sql: create index if not exists json_data_uuid on ${AQTS_SCHEMA_NAME}.json_data (partition_number, uuid);
+        - rollback: drop index if exists ${AQTS_SCHEMA_NAME}.json_data_uuid;
```
Comment (ayan-usgs) [inline liquibase/changeLogs/capture/indexes/jsonData.yml]: I'm wondering if we should also have indices for each of the partition tables on uuid... :thinking:

Comment (kkehl-usgs) [reply liquibase/changeLogs/capture/indexes/jsonData.yml]: @ayan-usgs 
It looks like line #12 does that.  I'm looking in pgadmin, and I see indexes on the partitions, like:

Tables->json_data->Partitions->json_data_august->indexes->json_data_august_partition_number_uuid_idx

Comment (ayan-usgs) [reply liquibase/changeLogs/capture/indexes/jsonData.yml]: Huh... I only saw the one on json_id. :man_shrugging: Alright then.

Comment (kkehl-usgs) [reply liquibase/changeLogs/capture/indexes/jsonData.yml]: Actually you are correct.  The way I had named it was confusing.  I have changed it so there are indexes on json_data_id and uuid on all partitions.

**Pull request merged**